src/xform_to_datalog/authorization_logic_datalog_facts_test.cc

```diff
@@ -0,0 +1,55 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+#include "src/xform_to_datalog/authorization_logic_datalog_facts.h"
+
+#include <cstdlib>
+#include <filesystem>
+#include <fstream>
+#include <vector>
+
+#include "absl/strings/str_split.h"
+#include "src/common/testing/gtest.h"
+#include "src/xform_to_datalog/authorization_logic_test.h"
+
+namespace fs = std::filesystem;
+
+namespace raksha::xform_to_datalog {
+
+class AuthorizationLogicDatalogFactsTest : public AuthorizationLogicTest {};
+
+TEST_F(AuthorizationLogicDatalogFactsTest, InvokesRustToolAndGeneratesOutput) {
+  for (const std::string& program : {"simple_auth_logic", "empty_auth_logic"}) {
+    const fs::path& test_data_dir = GetTestDataDir();
+    auto auth_facts = AuthorizationLogicDatalogFacts::create(test_data_dir, program);
+    ASSERT_TRUE(auth_facts.has_value());
+  
+    std::vector<std::string> actual_datalog =
+      absl::StrSplit(auth_facts->ToDatalog(), "\n", absl::SkipEmpty());
+    std::vector<std::string> expected_datalog =
+      ReadFileLines(test_data_dir / (program + ".dl"));
+
+    // Need to compare individual lines as the output order is non-deterministic.
+    ASSERT_THAT(actual_datalog,
+                testing::UnorderedElementsAreArray(expected_datalog));
+  }
```

Comment:
Rather than putting this in a loop, might be better to use `TEST_P` and have `AuthorizationLogicDatalogFactsTest` inherit from `TestWithParam`. That said, I see that there are only two elements there and you're using this in other places, so feel free to ignore this if you think the tradeoff would be uglier.

Reply:
I did consider it. Unfortunately, this relies on  some methods defined in `AuthorizationLogicTest` (which in turn inherits from `::testing::Testing`). I have filed #150 to clean it up. I will move to a parameterized test at that time.

Reply:
Good enough for me.